include/vcmi/Creature.h

```diff
@@ -10,15 +10,29 @@
 
 #pragma once
 
-#include "Entity.h"
+#include "FactionMember.h"
 
 VCMI_LIB_NAMESPACE_BEGIN
 
 class CreatureID;
 class ResourceSet;
 enum class EGameResID : int8_t;
 
-class DLL_LINKAGE Creature : public EntityWithNativeTerrain<CreatureID>
+/// Base class for creatures and battle stacks
+class DLL_LINKAGE ACreature: public AFactionMember
+{
+public:
+	bool isLiving() const; //non-undead, non-non living or alive
+	ui32 Speed(int turn = 0, bool useBind = false) const; //get speed (in moving tiles) of creature with all modificators
```

Comment:
class methods should start from lower case: https://wiki.vcmi.eu/Coding_guidelines#Casing

> Local variables and methods start with a lowercase letter and use the camel casing

Reply:
this is also is just moved from IBonusBearer, but I can change.

Reply:
done